httpapi_plugins/ftd.py

```diff
@@ -129,6 +129,31 @@ def refresh_token_payload(refresh_token):
             raise ConnectionError(
                 'Server returned response without token info during connection authentication: %s' % response)
 
+    def _lookup_login_url(self, payload):
+        """When token generation URL is not defined in the inventory we may try to find it on the fly.
+        :param payload: Token request payload
+        :type payload: dict
+        :return: token generation response
+        """
+        for url in self._get_known_token_paths():
+            try:
+                response = self._login_api_call(payload, url)
+            except ConnectionError:
```

Comment:
With current implementation, wrong username\password will also result in `Token generation API was not found` exception.  It would be nice to handle this situation separately. FTD returns 400 with a proper error message when bad credentials are given.

![screen shot 2019-03-05 at 2 30 15 pm](https://user-images.githubusercontent.com/4347159/53805664-789e8180-3f53-11e9-8f09-7f0a8a5f5208.png)

In this case, we should stop iteration and raise an exception saying that wrong credentials were supplied.

Reply:
fixed